### PR TITLE
Add a lint to warn against using of tokio::time::interval(_at)

### PR DESCRIPTION
**Pull request by tomaszklak**
This function return an interval that can cause huge number of ticks right after a long period of sleep.

### CODE REVIEW

**Review by gytsto — APPROVED** [review]
+1 lgtm

File: existing_lints/tokio_time_interval/ui/tokio_time_interval.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        fn interval(_: Duration) -> u32 {
+            42
+        }
+        let _interval = interval(D);
+    }
+
+    #[allow(tokio_time_interval)]
+    fn correct() {
+        let mut interval = tokio::time::interval(D);
+        interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
```
Comment (packgron) [inline existing_lints/tokio_time_interval/ui/tokio_time_interval.rs:29]: Would be hard to check if next line has missed tick setting?
I would say that resolving complex settup logic some code after settup would not be nice, but
ideomaticaly we should see set_missed_tick_behavour after interval creation, and in such cases it should no longer warn.
For more complex cases, we can leave allow

File: existing_lints/tokio_time_interval/src/tokio_time_interval.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    /// **Example:**
+    ///
+    /// If there is some periodic action that is triggered by the resulting Interval instance,
+    /// after a long period of sleep it will be triggered 'sleep_duration/interval_period' as fast
+    /// as possible until it is caught up in time to where it should be. Most of the time this is
+    /// not a desired behaviour. This default behaviour can be controlled with the 'set_missed_tick_behavior'
+    /// setter. In libtelio there is already a wrapper that returns correctly configured interval.
+    pub TOKIO_TIME_INTERVAL,
+    Warn,
+    "description goes here"
```
Comment (packgron) [inline existing_lints/tokio_time_interval/src/tokio_time_interval.rs:28]: Maybe a proper description would be nice?

File: existing_lints/tokio_time_interval/ui/tokio_time_interval.rs (hunk 1 above)
Comment (packgron) [reply existing_lints/tokio_time_interval/ui/tokio_time_interval.rs:29]: Ok I investigated a bit, and i guess the apis are not really designed for this at lest for an LatePass

Comment (tomaszklak) [reply existing_lints/tokio_time_interval/ui/tokio_time_interval.rs:29]: Yup, I considered a proper lint that would detect that the very next line is setting correct behaviour but I don't see a clear way to implement this without writing lots of code to manage state between different statements being visited.

File: existing_lints/tokio_time_interval/src/tokio_time_interval.rs (hunk 2 above)
Comment (tomaszklak) [reply existing_lints/tokio_time_interval/src/tokio_time_interval.rs]: Done 😄

**Review by packgron — APPROVED** [review]
+1.0

**Pull request merged**